library/src/main/java/com/getstream/sdk/chat/view/Dialog/ChannelMoreActionDialog.java

```diff
@@ -0,0 +1,141 @@
+import com.getstream.sdk.chat.viewmodel.ChannelListViewModel;
+
+public class ChannelMoreActionDialog extends Dialog {
+
+    Channel channel;
+    ChannelListViewModel viewModel;
+    Context context;
+    public ChannelMoreActionDialog(@NonNull Context context) {        
+        super(context, R.style.DialogTheme);
+        this.context = context;
+        Utils.hideSoftKeyboard((Activity) context);
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL,
+                WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL);
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_WATCH_OUTSIDE_TOUCH, WindowManager.LayoutParams.FLAG_WATCH_OUTSIDE_TOUCH);
+    }
+
+    public ChannelMoreActionDialog setChannelListViewModel(ChannelListViewModel viewModel) {
+        this.viewModel = viewModel;
+        init();
+        return this;
+    }
+
+
+    public ChannelMoreActionDialog setChannel(Channel channel) {
+        this.channel = channel;
+        init();
+        return this;
+    }
+
+
+    public void init() {
+        if (viewModel == null || channel == null)
+            return;
+
+        setContentView(R.layout.stream_dialog_channel_moreaction);
+        setCanceledOnTouchOutside(true);
+        LinearLayout ll_hide = findViewById(R.id.ll_hide);
+        LinearLayout ll_edit = findViewById(R.id.ll_edit);
+        LinearLayout ll_delete = findViewById(R.id.ll_delete);
+
+        ll_edit.setVisibility(canEditOrDeleteChannel() ? View.VISIBLE : View.GONE);
+        ll_delete.setVisibility(canEditOrDeleteChannel() ? View.VISIBLE : View.GONE);
+
+        ll_hide.setOnClickListener(view-> hideChannel());
+        ll_edit.setOnClickListener(view-> editChannel());
+        ll_delete.setOnClickListener(view-> deleteChannel());
+    }
+
+    private boolean canEditOrDeleteChannel() {
+        return channel.getCreatedByUser() != null && channel.getCreatedByUser().getId().equals(channel.getClient().getUserId());
+    }
+
+    private void hideChannel(){
+        dismiss();
+        viewModel.hideChannel(channel, new ResultCallback<Void, String>() {
+            @Override
+            public void onSuccess(Void aVoid) {
+                Utils.showMessage(context, "Hidden successfully!");
+            }
+
+            @Override
+            public void onError(String s) {
+                Utils.showMessage(context, s);
+            }
+        });
+    }
+
+    private void showChannel(){
+        dismiss();
+        viewModel.showChannel(channel, new ResultCallback<Void, String>() {
+            @Override
+            public void onSuccess(Void aVoid) {
+                Utils.showMessage(context, "Shown up successfully!");
+            }
+
+            @Override
+            public void onError(String s) {
+                Utils.showMessage(context, s);
+            }
+        });
+    }
+
+    private void editChannel(){
+        dismiss();
+
+    }
+
+    private void deleteChannel(){
```

Comment:
we should remove this; in most cases channels are deleted server-side (deleting a channel removes it for all users on the channel)